pkg/channel/message_receiver.go

```diff
@@ -211,10 +218,10 @@ func (r *MessageReceiver) fromHTTPHeaders(headers http.Header) map[string]string
 
 // ParseChannel converts the channel's hostname into a channel
 // reference.
-func ParseChannel(host string) (channel.ChannelReference, error) {
-	chunks := strings.Split(host, ".")
+func ParseChannel(host url.URL) (channel.ChannelReference, error) {
+	chunks := strings.Split(host.Path, ".")
 	if len(chunks) < 2 {
-		return channel.ChannelReference{}, fmt.Errorf("bad host format '%s'", host)
+		return channel.ChannelReference{}, fmt.Errorf("bad host format %v", host)
```

Comment:
Should be `%q` to preserve the original intention of quoting the output.

Reply:
+1

Reply:
The type of host has changed, so I get this error with %q: Errorf format %q has arg host of wrong type net/url.URL

I can manually add in quotes though!

Reply:
%q invokes String(), so you can use it safely.